genesyscloud/telephony_providers_edges_edge_group/data_source_genesyscloud_telephony_providers_edges_edge_group.go

```diff
@@ -11,31 +11,28 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/mypurecloud/platform-client-sdk-go/v121/platformclientv2"
 )
 
 func dataSourceEdgeGroupRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	sdkConfig := m.(*gcloud.ProviderMeta).ClientConfig
-	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+	edgeGroupProxy := getEdgeGroupProxy(sdkConfig)
 
 	name := d.Get("name").(string)
 	managed := d.Get("managed").(bool)
 
 	return gcloud.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
-		for pageNum := 1; ; pageNum++ {
-			const pageSize = 100
-			edgeGroup, _, getErr := edgesAPI.GetTelephonyProvidersEdgesEdgegroups(pageSize, pageNum, name, "", managed)
 
-			if getErr != nil {
-				return retry.NonRetryableError(fmt.Errorf("Error requesting edge group %s: %s", name, getErr))
-			}
+		edgeGroups, getErr := edgeGroupProxy.getEdgeGroupByName(ctx, name, managed)
 
-			if edgeGroup.Entities == nil || len(*edgeGroup.Entities) == 0 {
-				return retry.RetryableError(fmt.Errorf("No edge group found with name %s", name))
-			}
+		if getErr != nil {
+			return retry.NonRetryableError(fmt.Errorf("Error requesting edge group %s: %s", name, getErr))
+		}
 
-			d.SetId(*(*edgeGroup.Entities)[0].Id)
-			return nil
+		if edgeGroups == nil || len(*edgeGroups) == 0 {
```

Comment:
this should be `if retryable`, `getEdgeGroupByName` should return a boolean indicating if you should retry